include/ptp_clock.h

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _PTP_CLOCK_H_
+#define _PTP_CLOCK_H_
+
+#include <stdint.h>
+#include <device.h>
+#include <misc/util.h>
+#include <net/ptp_time.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* Name of the PTP clock driver */
+#if !defined(PTP_CLOCK_NAME)
+#define PTP_CLOCK_NAME "PTP_CLOCK"
+#endif
+
+struct ptp_clock_driver_api {
+	int (*set)(struct device *dev, struct net_ptp_time *tm);
```

Comment:
Ok the orginial patches were really messy. 

Grab from #6963 the net_ptp_time.h (and remove it from there). 
Insert the struct declaration in this file, but call it struct ptp_time

(If the name net_ptp_time is used a lot in gptp code or eles, just do a typedef as an alias somewhere relevant there)

Reply:
I rather keep the ptp_clock device as is and not introduce the ptp_time into this PR as that time definition has nothing to do with the device driver.

Reply:
it's using it and it seems to be the only type of data it understands, so it has to do with a bit still. (At least getting ptp time declared in the other patch that does net if cb, it not relevant)